src/FileReader_Core.cpp

```
@@ -363,22 +519,24 @@ bool FileReader::preprocessFileContent(const std::string& file_content) {
 			std::istringstream iss(line);
 			iss.seekg("#define"sv.length());
 			handleDefine(iss);
 			append_line_with_tracking("");  // Preserve line numbering
 		} else if (line.find("#ifdef", 0) == 0) {
-			std::istringstream iss(line);
-			iss.seekg("#ifdef"sv.length());
-			std::string symbol;
-			iss >> symbol;
+			PreprocessTimer timer(PreprocessTimingCategory::conditional, false);
+			std::string_view sv(line);
+			sv.remove_prefix("#ifdef"sv.length());
```

Cppcheck (2.20.0)

[error] Syntax Error

(internalAstError)